File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap13/CallbackSearchOrder.py
#!/usr/bin/env python

### CallbackSearchOrder.py

from Tkinter import *                    

mw = Tk()                                                            #(A)
b = Button( mw )                                                     #(B)
b.pack()                                                             #(C)

mw.bind( '<Button>', lambda e: foo_tops() )                          #(D)
b.bind( '<Button>', lambda e: foo_butt() )                           #(E)
b.bind_class( 'Button', '<Button-1>', lambda e: foo_Butt() )         #(F)
b.bind_all( '<Button>', lambda e: foo_App() )                        #(G)

def foo_tops():  print "callback for Toplevel instance invoked"      #(H)
def foo_butt(): print "callback for Button instance invoked"         #(I)
def foo_Butt(): print "callback for Button class invoked"            #(J)
def foo_App(): print "callback for Application level invoked"        #(K)

mainloop()                                                           #(L)

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap13/CallbackSearchOrder.pl
#!/usr/bin/perl -w

### CallbackSearchOrder.pl

use strict;
use Tk;

my $mw = MainWindow->new();                                          #(A)
my $b = $mw->Button()->pack();                                       #(B)

sub foo_tops { print "callback for Toplevel instance invoked\n" }    #(C)
$mw->bind( '<Button>', \&foo_tops);                                  #(D)

sub foo_Butt { print "callback for Button class invoked\n" }         #(E)
$b->bind( 'Tk::Button', '<Button-1>', \&foo_Butt );                  #(F)

sub foo_butt { 
    my $tag = shift;
    print "callback for Button instance invoked\n";
    $tag->break;
}                                                                    #(G)

$b->bind( '<Button>', \&foo_butt);                                   #(H)

sub foo_App { print "callback for Application level invoked\n" }     #(I)
$b->bind( 'all', '<Button>', \&foo_App);                             #(J)

my @tags = $b->bindtags;
$b->bindtags( [@tags[1, 0, 2, 3]]);

MainLoop;                                                            #(K)
